pkg/vsphere/vm/vm.go

```diff
@@ -281,7 +284,37 @@ func (vm *VirtualMachine) DeleteExceptDisks(ctx context.Context) (*object.Task,
 		return nil, err
 	}
 
-	return vm.Destroy(op)
+	info, err := vm.WaitForResult(op, func(ctx context.Context) (tasks.Task, error) {
+		return vm.Destroy(ctx)
+	})
+
+	if err == nil {
+		return info, nil
+	}
+
+	if isMethodDisabledError(ctx, err) {
```

Comment:
reverse the test - it's clearer to return early:
```
if !isMethodDisabledError(ctx, err) {
  return nil, err
}
```
you may even be able to combine it with the test above:
```
if err == nil || !isMethodDisabledError(ctx, err) {
   return info, err
}
```